src/server_manager/web_app/ui_components/outline-server-settings.ts

```diff
@@ -441,7 +441,7 @@ Polymer({
   _defaultDataLimitEnabledChanged(e: CustomEvent) {
     const wasDataLimitEnabled = this.isDefaultDataLimitEnabled;
     const isDataLimitEnabled = e.detail.value === 'enabled';
-    if (isDataLimitEnabled === undefined || wasDataLimitEnabled === undefined) {
+    if (typeof isDataLimitEnabled === undefined || typeof wasDataLimitEnabled === undefined) {
```

Comment:
First, there's nothing wrong with testing against undefined with `===` for types that are optional. That is clearer than typeof. The problem is the typing of the variables.

Another issue here is that `isDataLimitEnabled` is guaranteed to be boolean, so this test is bad.

This code would make the types clearer and the code easier to understand:
```ts
if (e.detail?.value === undefined) {
  return;
}
const isDataLimitEnabled = e.detail.value === 'enabled';  // guaranteed to be real Boolean
// This is actually odd, I don't think we need this test. Should probably remove.
// if (this.isDefaultDataLimitEnabled === undefined) {
//  return;
// }
if (isDataLimitEnabled === this.isDefaultDataLimitEnabled) {
  return;
}
this.isDefaultDataLimitEnabled = isDataLimitEnabled;
if (isDataLimitEnabled) {
  this._requestSetDefaultDataLimit();
} else {
  this.fire('RemoveDefaultDataLimitRequested');
}
```

Reply:
LGTM!